shared/actions/chat2/index.js

```diff
@@ -1271,64 +1271,103 @@ const messageRetry = (action: Chat2Gen.MessageRetryPayload, state: TypedState) =
+      ephemeralLifetime
+    )
 
-  logger.info('[MessageSend]', 'non-empty text?', text.stringValue().length > 0)
+    const addMessage = (p, response) => [
+      Saga.put(
+        Chat2Gen.createMessagesAdd({
+          context: {type: 'sent'},
+          messages: [newMsg],
+        })
+      ),
+      Saga.callUntyped(function() {
+        response && response.result()
+      }),
+    ]
+    const onShowConfirm = () => {
+      stellarConfirmPaymentModalShown = true
+      return [
+        Saga.put(Chat2Gen.createClearPaymentConfirmInfo()),
+        Saga.put(
+          RouteTreeGen.createNavigateAppend({
+            path: ['paymentsConfirm'],
+          })
+        ),
+      ]
+    }
+    const onHideConfirm = () => {
+      if (!stellarConfirmPaymentModalShown) {
+        return
+      }
+      stellarConfirmPaymentModalShown = false
+      return Saga.put(navigateUp())
+    }
+    const onDataConfirm = ({summary}, response) => {
+      stellarConfirmWindowResponse = response
+      return Saga.put(Chat2Gen.createSetPaymentConfirmInfo({summary}))
+    }
+    const onDataError = ({message}, response) => {
+      stellarConfirmWindowResponse = response
+      return Saga.put(Chat2Gen.createSetPaymentConfirmInfoError({error: message}))
+    }
+    yield RPCChatTypes.localPostTextNonblockRpcSaga({
+      customResponseIncomingCallMap: {
+        'chat.1.chatUi.chatPostReadyToSend': addMessage,
+        'chat.1.chatUi.chatStellarDataConfirm': onDataConfirm,
+        'chat.1.chatUi.chatStellarDataError': onDataError,
+      },
+      incomingCallMap: {
+        'chat.1.chatUi.chatStellarDone': onHideConfirm,
+        'chat.1.chatUi.chatStellarShowConfirm': onShowConfirm,
+      },
+      params: {
+        ...ephemeralData,
+        body: text.stringValue(),
+        clientPrev,
+        conversationID: Types.keyToConversationID(conversationIDKey),
+        identifyBehavior: getIdentifyBehavior(state, conversationIDKey),
+        outboxID,
+        tlfName,
+        tlfPublic: false,
+      },
+      waitingKey: Constants.waitingKeyPost,
+    })
 
-  // We need to put an addMessage ahead of postText in case we get new activity on that outboxID before the
-  // the action to add the pending message fires. This would cause a pending message to be stuck
-  // (with a duplicate sent message in there too).
-  //
-  // We put the addMessage on the back in case the service provides chat thread data in between the
-  // addMessage and postText action. upgradeMessage should be a no-op in the case that the message
-  // that is in the store on the outboxID has been sent.
-  return Saga.sequentially([addMessage, postText, addMessage])
-}
+    // Do some logging to track down the root cause of a bug causing
+    // messages to not send. Do this after creating the objects above to
+    // narrow down the places where the action can possibly stop.
+    logger.info('[MessageSend]', 'non-empty text?', text.stringValue().length > 0)
```

Comment:
likely want to clear stellarConfirmWindowResponse here too

Reply:
It gets cleared right before we show it.

Reply:
but better to not let it lying around right. seems more error prone